components/menu/menu.js

```diff
@@ -260,6 +279,11 @@ class Menu extends ThemeMixin(HierarchicalViewMixin(LitElement)) {
 		if (returnItem) returnItem.setAttribute('text', this.label);
 	}
 
+	_onFocusOut(e) {
```

Comment:
It might not affect the behaviour negatively, but `focusout` bubbles so in the case of a nested menu's menu-item, I think we'd run this for every menu the event bubbles past. I this case, when focus is lost from the deepest menu-item, both menu components would apply this change to it.
```
menu
    menu-item
        menu
            menu-item
```

Reply:
It looks like we'd be leaving `tabindex="0"` on the first item. Does this mean that if a user pressed [shift-Tab] on a subsequent item, focus would return to the first item instead of soft-closing the dropdown like we'd normally expect? i.e. we could have something like this rendered...
```
menu-item tabindex="0"
menu-item tabindex="-1"
menu-item tabindex="0"
```

Reply:
That does mean that for the `shift-tab` behaviour it would re-focus on the first item, though that's also the current behaviour (which to me I think makes sense but let me know if we should be changing that). That's just with testing on the first dropdown menu [here](https://live.d2l.dev/BrightspaceUI/core/components/dropdown/demo/dropdown-menu.html) so let me know if there's a different one that has a different behaviour that I'm missing.

Reply:
Might be worth checking the Aria guidelines and/or Jeff for that. My expectation is that it would move to the previous focusable before the menu, but I could be wrong.

Reply:
I've message Jeff about this. It would be potentially 2 changes to existing functionality:
1. shift-tab would move to first focusable outside of menu rather than first menu item (would this be the opener)
2. (possibly) shift-tab to the opener closes the dropdown

I believe we'd want to handle that outside of this PR since these are separate functionality changes.

Reply:
Jeff's thoughts:
1. shift-tab should move focus to the opener and close the menu
2. Home/end should jump to first/last menu item (this is also in the pattern and not something that we currently implement)

Let me know thoughts on this. I'll spin up a couple more stories for these pieces.

Reply:
Just a note about "shift-tab should move focus to the opener and close the menu"... `d2l-menu` might not necessarily be in a dropdown - it is perfectly valid for it to be used outside this context, where there is no opener. I don't _think_ that matters so far, but just wanted to call out it out in case it does.

Aside from that, do we also need to check for the other menu-item roles?

Reply:
Yup for sure! I'll include that in the story just to be clear.

Reply:
Both of those points make sense to me. I'll add those details to the stories.

Reply:
Ah sorry - a different one. Since this is keeping the same behaviour as before (other than handling forward tab) I opened separate stories for the other 2 changes.